tests/cypress/e2e/results.cy.js

```diff
@@ -77,493 +78,595 @@ config?.pages?.forEach((page, _i) => {
+			beforeEach("reset", function () {
+				cy.snapController().then(({store, urlManager}) => {
+					urlManager.set({}).go();
+					cy.wait(200);
 				});
-			});
+			}); 
+		
+			describe('Pagination', () => {
 
-			it('can go back to the first page', function () {
-				if (!config?.selectors?.pagination?.prev && !config?.selectors?.pagination?.page) this.skip();
+				it("loads next page when scrolled while using Infinite Scroll", function () {
+					if (!config.selectors.pagination.infinite) this.skip();
+					cy.snapController().then(({store }) => {
 
-				if (config?.selectors?.pagination?.prev) {
-					cy.get(`${config.selectors.pagination.prev}`)
-						.first()
-						.should('exist')
-						.as('prev')
-						.invoke('attr', 'href')
-						.then(function (href) {
-							if (href) {
-								cy.get('@prev').click({ force: true });
-							} else {
-								cy.get('@prev').find('a, button').click({ force: true });
-							}
-						});
-				} else if (config?.selectors?.pagination?.page) {
-					cy.get(config.selectors.pagination.page)
-						.eq(0)
-						.should('exist')
-						.as('page1')
-						.invoke('attr', 'href')
-						.then(function (href) {
-							if (href) {
-								cy.get('@page1').click({ force: true });
-							} else {
-								cy.get('@page1').find('a, button').click({ force: true });
-							}
-						});
-				}
+						expect(store.pagination.page).to.equal(1);
+						cy.get(config.selectors.results?.productWrapper).should('exist').should('have.length', store.pagination.pageSize);
+
+					});
+
+					if (config.selectors.pagination.loadMoreButton) {
+						cy.get(config.selectors.pagination.loadMoreButton).should('exist').first().click({force: true});
+					}else {
+						cy.window().scrollTo('bottom', { ensureScrollable: false });
+					}
+
+					cy.snapController().then(({ store }) => {
+						expect(store.pagination.page).to.equal(2);
+						cy.get(config.selectors.results?.productWrapper).should('exist').should('have.length', store.pagination.pageSize * 2);
+
+					});
+
+					if (config.selectors.pagination.loadMoreButton) {
+						cy.get(config.selectors.pagination.loadMoreButton).should('exist').first().click({force: true});
+					}else {
+						cy.window().scrollTo('bottom', { ensureScrollable: false });
+					}
+
+					cy.snapController().then(({ store }) => {
+						expect(store.pagination.page).to.equal(3);
+						cy.get(config.selectors.results?.productWrapper).should('exist').should('have.length', store.pagination.pageSize * 3);
+						cy.window().scrollTo('top', { ensureScrollable: false });
+
+					});
 
-				cy.snapController().then(({ store }) => {
-					expect(store.pagination.page).to.equal(1);
 				});
-			});
 
-			it('can go to the third page', function () {
-				if (!config?.selectors?.pagination?.page) this.skip();
+				it('can navigate to the second page', function () {
+					if (!config?.selectors?.pagination?.next && !config?.selectors?.pagination?.page) this.skip();
```

Comment:
Should it skip these pagination tests when infinite is set? I realize they could just empty out those selectors alternatively. But maybe allow for both?